sdk/storage/MigrationGuide.md

```diff
@@ -0,0 +1,477 @@
+# Migration Guide: From Azure Storage Legacy CPP SDK to Azure Stroage CPP SDK
+
+This guide intends to assist customers in migrating from legacy versions of the Azure Storage C++ library for Blobs to version 12.
+While this guide is generally applicable to older versions of the SDK, it was written with v7.5 in mind as the starting point.
+It will focus on side-by-side comparisons for similar operations between the [Track1 SDK](https://github.com/Azure/azure-storage-cpp) and [Track2 SDK](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/storage).
+Familiarity with the legacy client library is assumed. For those new to the Azure Storage Blobs client library for C++, please refer to the [Quickstart](https://docs.microsoft.com/en-us/azure/storage/blobs/quickstart-blobs-c-plus-plus) for the v12 library rather than this guide.
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+- [General changes](#general-changes)
+  - [Package and namespaces](#package-and-namespaces)
+  - [Authentication](#authentication)
+  - [Client structure](#client-structure)
+- [Migration samples](#migration-samples)
+  - [Creating a Container](#creating-a-container)
+  - [Uploading Blobs to a Container](#uploading-blobs-to-a-container)
+  - [Downloading Blobs from a Container](#downloading-blobs-from-a-container)
+  - [Listing Blobs in a Container](#listing-blobs-in-a-container)
+  - [Managing Blob Metadata](#managing-blob-metadata)
+  - [Content Hashes](#content-hashes)
+  - [Resiliency](#resiliency)
+- [Additional information](#additional-information)
+
+## Migration benefits
+
+To understand why we created our version 12 client libraries, you may refer to the Tech Community blog post, [Announcing the Azure Storage v12 Client Libraries](https://techcommunity.microsoft.com/t5/azure-storage/announcing-the-azure-storage-v12-client-libraries/ba-p/1482394) or refer to our video [Introducing the New Azure SDKs](https://aka.ms/azsdk/intro).
+
+Included are the following:
+- Thread-safe synchronous APIs
+- Improved performance
+- Consistent and idiomatic code organization, naming, and API structure, aligned with a set of common guidelines
+- The learning curve associated with the libraries was reduced
```

Comment:
Full stop for each line?
Also, could add `More comprehensive feature parity.`, `Removed dependencies on CPPRestSDK and Boost.`, `Better supportability in feature requests and issue fixes.` which are also selling points for track 2 SDK.

Reply:
I assume these are benefits of v12 client sdks, not only about C++ SDK

Reply:
I copied most of this doc from .Net

Reply:
`Advanced request customization with pipeline framework.` is another selling point.

Reply:
This is migration guide, not advertisement. I don't think most of customers would care advanced scenarios.

Reply:
If we have a migration guide, it would be the first thing to look at if customer is considering migration. I think it should at least make sure the advantages are at least stated.

Reply:
respectfully disagree.